### PR TITLE
Building workflows

### PR DESCRIPTION
**Pull request by hestonhoffman**
Documentation for Workflows: 
- Trigger a workflow
- Build a workflow
---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

**Comment by hestonhoffman** [comment]
Editorial review: https://datadoghq.atlassian.net/browse/DOCS-4165

File: content/en/workflows/trigger.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  text: "Find out more about setting up a dashboard"
+- link: "security_platform/explorer"
+  tag: "Documentation"
+  text: "Find out more about Security Signals"
+- link: "monitors"
+  tag: "Documentation"
+  text: "Find out more about Monitors"
+---
+
+You can trigger a workflow manually, on a schedule, or using a variety of automated methods such as a Datadog Monitor, or a webhook. Before you can trigger a workflow using automation, you must associate the workflow with a unique service account. For more information on setting up a service account for your workflows, see [Set up workflows][1].
```
Comment (natashadada) [inline content/en/workflows/trigger.md:17]: We no longer support webhooks but we do support security signal notification rules.

Comment (hestonhoffman) [reply content/en/workflows/trigger.md:17]: Ah thanks! I forgot to remove the webhook bit

File: content/en/workflows/build.md [hunk 2]
```diff
@@ -3,7 +3,126 @@ title: Build Workflows
...
+            "env": "default",
+            "connections": []
+        }
+    ]
+}
+{{< /code-block >}}
+
+## Context variables
+
+Creating useful workflows sometimes necessitates passing data from one step to another, or configuring steps that act on data that comes from the workflow's trigger source. You can perform this kind of data interpolation with context variables.
```
Comment (urseberry) [inline content/en/workflows/build.md:84]:
```suggestion
Creating useful workflows sometimes necessitates passing data from one step to another, or configuring steps that act on data from the workflow's trigger source. You can perform this kind of data interpolation with context variables.
```

File: content/en/workflows/build.md [hunk 3]
```diff
@@ -3,7 +3,126 @@ title: Build Workflows
...
+### Workflow variables
+
+All workflows have three standard variables:
+- `WorkflowName`: The name of the workflow. Accessed with `{{ WorkflowName }}`.
+- `WorkflowId`: The workflow ID. Accessed with `{{ WorkflowId }}`.
+- `InstanceId`: Each workflow run receives a unique instance ID. Access the instance ID with `{{ InstanceId }}`.
+
+### Step output variables
+
+Some steps create outputs that are available to subsequent steps in a workflow. Access a step variable with the syntax: `Steps.<step_name>.<variable>`. For example, to retrieve the pull request status variable (`state`) from the GitHub pull request status step (`Get_pull_request_status`), you'd use the following context variable:
```
Comment (urseberry) [inline content/en/workflows/build.md:107]: Is there a way to list the step variables?

File: content/en/workflows/build.md [hunk 4]
```diff
@@ -3,7 +3,126 @@ title: Build Workflows
...
+
+```
+{{ Steps.Get_pull_request_status.state }}
+```
+
+### Trigger variables
+
+You can pass trigger variables into a workflow as inputs. Workflows accept a JSON object of comma-separated key-value pairs. Access a trigger variable in workflow steps using the syntax `{{ Trigger.key }}`. For example, to access the trigger variable `{ "user": "Bits" }`, use ` {{ Trigger.user }}` in the step.
```
Comment (urseberry) [inline content/en/workflows/build.md:115]: Same thing as above - is there a way to get a list of the available trigger variables for a step?

**Review by urseberry — COMMENTED** [review]
Completed the review for the build page. Will review the trigger page next.

File: content/en/workflows/trigger.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+1. Under **Execution parameters**, enter the required trigger variables for your workflow.
+1. Click **Run** to run the workflow.
+
+## Trigger a workflow from a Monitor
+
+To trigger a workflow from a Monitor:
+1. On the workflow canvas, click **Add an Automated Trigger** and select **@mention**.
+1. Next to **@workflow-**, enter a mention name for the trigger. Your mention name must be unique.
+1. Save your Workflow.
+1. Find the monitor you'd like to use to trigger the workflow and edit it, or create a new monitor.
```
Comment (urseberry) [inline content/en/workflows/trigger.md:47]: Can you be more clear leaving Workflows and navigating to monitors here? Maybe add a step like "Navigate to the monitors list" or something.

File: content/en/workflows/trigger.md [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+1. In the message section, add the full workflow mention name:
+   - The mention name should start with `@workflow-`. For example, `@workflow-my-workflow`
+   - To pass trigger variables into the workflow, use a comma-separated list with the syntax `@workflow-name(key=value, key=value)`. For example, `@workflow-my-workflow(name="Bits", alert_threshold=threshold)`
+1. Save the monitor.
+
+{{< img src="workflows/monitor-trigger.png" alt="Add a monitor trigger to the message section of a Monitor" >}}
+
+Each time the monitor threshold is hit, the monitor triggers a workflow run.
```
Comment (urseberry) [inline content/en/workflows/trigger.md:55]:
```suggestion
Each time the monitor threshold is hit, the monitor triggers a workflow run.

```
Adding a line for visual clarity when people view the markdown source.

**Review by urseberry — COMMENTED** [review]
Very detailed. Very nice!

File: content/en/workflows/trigger.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+  text: "Find out more about setting up a dashboard"
+- link: "security_platform/explorer"
+  tag: "Documentation"
+  text: "Find out more about Security Signals"
+- link: "monitors"
+  tag: "Documentation"
+  text: "Find out more about Monitors"
+---
+
+You can trigger a workflow manually, on a schedule, or using a variety of automated methods such as a Datadog Monitor, or a Security Signal Notification Rule. Before you can trigger a workflow using automation, you must associate the workflow with a unique service account. For more information on setting up a service account for your workflows, see [Set up workflows][1].
```
[the remark this answers is not in the document]
Comment (hestonhoffman) [reply content/en/workflows/trigger.md:17]: Okay, let's add this in a follow-up PR

**Review by urseberry — APPROVED** [review]

File: content/en/workflows/build.md [hunk 8]
```diff
@@ -3,7 +3,141 @@ title: Build Workflows
...
+You can create workflows or edit existing workflows from the [**Workflows** page][1] in Datadog. The Workflows page lists existing workflows together with each workflow's author and the dates that each workflow was last modified and executed.
+- Hover over a workflow for the options to delete or clone the workflow.
+- Toggle **My workflows** if you want to see only workflows that you created.
+
+## Build a workflow from a blueprint
+
+1. Click the **Blueprints** tab.
+1. If desired, use the search bar to narrow the list of blueprints by name, category, or integration.
+1. Find the blueprint you would like to use, and click on it. The workflow canvas appears.
+1. Click the pencil icon to update the workflow name. If desired, type a new name in the text box and click **Save**.
```
Comment (urseberry) [inline content/en/workflows/build.md:20]: This line maybe needs the same edit you made to my other PR? Add the tooltip text (**Edit**) in bold?

**Pull request merged**